### DISCUSSION TITLE
Integration of plugins in a holistic emulator

### DISCUSSION THREAD

**Discussion by Papermanzero** [simple64/simple64#450] [General]
In general all emulators are going away from plugin systems to create a holistic emulator which is much easier to maintain and optimise.
Are you going to do that with simple64 as well?

it is really the right approach and the compatibility of simple is very high.

**Comment by loganmc10** [comment]
What is the request exactly? To change the way the code is linked together? Simple64 uses parallel-rdp for graphics and parallel-rsp for emulation of the RSP, but they aren't really "plugins" since you can't swap them out.

**Reply by Papermanzero** [reply]
That’s exactly what I mean. The codebase is still separate (Parallel RDP/RSP<-> Mupen Core).
The question is if it makes sense to integrate everything into one holistic codebase (like what pscx2 did for example). 

Although you cannot exchange the plugins the code is still relying on the development of Parallel and Mupen.

the request or discussion what I wanted to started is if it makes sense to integrate everything into one codebase and develop simple64 independently of mupen64 and/or parallel.

That would also mean that you can create features independently.

please don’t get me wrong. I am not requesting to integrate everything into one code stack. It is a just a discussion what I wanted to start out of curiosity and to maybe bring up some ideas.

**Reply by Papermanzero** [reply]
https://github.com/PCSX2/pcsx2/pull/4436

**Comment by loganmc10** [comment]
I don't see this happening for simple64. I am not the author of parallel-rsp or parallel-rdp, and I don't plan on maintaining them. It is actually easier the way it is now to pull in updates directly from the original repositories, rather than try to maintain my own version.
